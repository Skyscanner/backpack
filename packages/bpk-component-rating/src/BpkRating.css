@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-rating{display:flex}.bpk-rating--vertical{flex-direction:column}.bpk-rating__component{justify-content:center;align-items:center;border-radius:50%;color:#fff}.bpk-rating--base-rating{display:flex;width:3rem;height:3rem}.bpk-rating--lg-rating{display:flex;width:3.75rem;height:3.75rem}.bpk-rating--sm-rating{display:flex;width:2.25rem;height:2.25rem}.bpk-rating__text-wrapper{display:flex;flex-direction:column;justify-content:center}.bpk-rating__text-wrapper--horizontal{padding-left:.75rem}.bpk-rating__text-wrapper--vertical{padding-top:.375rem}.bpk-rating__text--horizontal{text-align:left}.bpk-rating__text--vertical{align-self:center;text-align:center}.bpk-rating--high-rating{align-self:center;background-color:#00a698;background-color:var(--bpk-rating-high-color, #00a698)}.bpk-rating--medium-rating{align-self:center;background-color:#ff9400;background-color:var(--bpk-rating-medium-color, #ff9400)}.bpk-rating--low-rating{align-self:center;background-color:#d1435b;background-color:var(--bpk-rating-low-color, #d1435b)}
