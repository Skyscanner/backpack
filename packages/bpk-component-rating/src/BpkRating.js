/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow strict */

import React, { type Node } from 'react';
import PropTypes from 'prop-types';
import clamp from 'lodash.clamp';

import { cssModules } from '../../bpk-react-utils';
import BpkText, { TEXT_STYLES } from '../../bpk-component-text';

import STYLES from './BpkRating.module.scss';
import { RATING_SIZES, RATING_SCALES } from './common-types';

const getClassName = cssModules(STYLES);

const getMinValue = () =>
  // Currently the min value is zero no matter what scale is used.
  // If this ever changes, this function should be changed to return
  // different values based on the rating scale.
  0;

const getMaxValue = (ratingScale) => {
  switch (ratingScale) {
    case RATING_SCALES.zeroToFive:
      return 5;
    default:
      return 10;
  }
};

type Props = {
  ariaLabel: string,
  className: ?string,
  ratingScale: $Values<typeof RATING_SCALES>,
  showScale: ?boolean,
  size: $Values<typeof RATING_SIZES>,
  subtitle: ?string,
  title: string | Node,
  value: string | number,
};

const BpkRating = (props: Props) => {
  const {
    ariaLabel,
    className,
    ratingScale,
    showScale,
    size,
    subtitle,
    title,
    value,
    ...rest
  } = props;

  const classNames = getClassName(
    'bpk-rating',
    className,
    size === RATING_SIZES.large && subtitle && 'bpk-rating--large',
  );
  const valueStyles = getClassName('bpk-rating__value');
  const scaleStyles = getClassName('bpk-rating__scale');
  const textWrapperStyles = getClassName(
    'bpk-rating__text-wrapper',
    size === RATING_SIZES.large && 'bpk-rating__text-wrapper--large',
  );
  const titleStyles = getClassName(
    size === RATING_SIZES.large && 'bpk-rating__title--large',
  );
  const subtitleStyles = getClassName(
    'bpk-rating__subtitle',
    size === RATING_SIZES.large && 'bpk-rating__subtitle--large',
  );

  let valueTextSize = TEXT_STYLES.label1;
  let scaleTextSize = TEXT_STYLES.caption;
  let titleTextSize = TEXT_STYLES.label1;
  let subtitleTextSize = TEXT_STYLES.caption;

  if (size === RATING_SIZES.large) {
    valueTextSize = TEXT_STYLES.hero5;
    scaleTextSize = TEXT_STYLES.bodyDefault;
    titleTextSize = TEXT_STYLES.heading5;
    subtitleTextSize = TEXT_STYLES.bodyDefault;
  }

  const minValue = getMinValue();
  const maxValue = getMaxValue(ratingScale);

  let adjustedValue = value;

  if (adjustedValue >= maxValue || adjustedValue <= minValue) {
    adjustedValue = clamp(adjustedValue, minValue, maxValue);
  }

  return (
    // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
    <div className={classNames} aria-label={ariaLabel} role="figure" {...rest}>
      <BpkText
        textStyle={valueTextSize}
        tagName="span"
        className={valueStyles}
        aria-hidden="true"
      >
        {adjustedValue}

        {showScale && (
          <BpkText
            textStyle={scaleTextSize}
            className={scaleStyles}
            tagName="span"
            aria-hidden="true"
          >
            /{maxValue}
          </BpkText>
        )}
      </BpkText>

      <div className={textWrapperStyles}>
        <BpkText
          textStyle={titleTextSize}
          className={titleStyles}
          tagName="span"
          aria-hidden="true"
        >
          {title}
        </BpkText>

        {subtitle && (
          <BpkText
            className={subtitleStyles}
            textStyle={subtitleTextSize}
            tagName="span"
            aria-hidden="true"
          >
            {subtitle}
          </BpkText>
        )}
      </div>
    </div>
  );
};

BpkRating.propTypes = {
  ariaLabel: PropTypes.string.isRequired,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  className: PropTypes.string,
  ratingScale: PropTypes.oneOf(Object.keys(RATING_SCALES)),
  size: PropTypes.oneOf(Object.keys(RATING_SIZES)),
  subtitle: PropTypes.string,
  showScale: PropTypes.bool,
};

BpkRating.defaultProps = {
  className: null,
  ratingScale: RATING_SCALES.zeroToFive,
  size: RATING_SIZES.base,
  subtitle: null,
  showScale: true,
};

export default BpkRating;
