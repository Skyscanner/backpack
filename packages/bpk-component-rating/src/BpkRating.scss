/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~bpk-mixins';

.bpk-rating {
  display: flex;

  &--vertical {
    flex-direction: column;
  }

  &__component {
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    color: $bpk-color-white;
  }

  &--base-rating {
    display: flex;
    width: $bpk-rating-base-width;
    height: $bpk-rating-base-height;
  }

  &--lg-rating {
    display: flex;
    width: $bpk-rating-lg-width;
    height: $bpk-rating-lg-height;
    font-size: $bpk-font-size-lg;
  }

  &--sm-rating {
    display: flex;
    width: $bpk-rating-sm-width;
    height: $bpk-rating-sm-width;
    font-size: $bpk-font-size-sm;
  }

  &__text-wrapper {
    display: flex;
    flex-direction: column;
    justify-content: center;

    &--horizontal {
      padding-left: $bpk-spacing-sm;
    }

    &--vertical {
      padding-top: $bpk-spacing-xs;
    }
  }

  &__text {
    &--horizontal {
      text-align: left;
    }

    &--vertical {
      align-self: center;
      text-align: center;
    }
  }

  &--high-rating {
    align-self: center;

    @include bpk-themeable-property(background-color, --bpk-rating-high-color, $bpk-color-monteverde);
  }

  &--medium-rating {
    align-self: center;

    @include bpk-themeable-property(background-color, --bpk-rating-medium-color, $bpk-color-kolkata);
  }

  &--low-rating {
    align-self: center;

    @include bpk-themeable-property(background-color, --bpk-rating-low-color, $bpk-color-panjin);
  }
}
