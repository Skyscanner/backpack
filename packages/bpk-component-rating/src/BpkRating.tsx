/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import PropTypes from 'prop-types';
import type { ReactNode } from 'react';

import clamp from 'lodash.clamp';

import BpkText, { TEXT_STYLES } from '../../bpk-component-text';
import { cssModules } from '../../bpk-react-utils';

import { RATING_SIZES, RATING_SCALES } from './common-types';

import STYLES from './BpkRating.module.scss';

const getClassName = cssModules(STYLES);

const getMinValue = () =>
  // Currently the min value is zero no matter what scale is used.
  // If this ever changes, this function should be changed to return
  // different values based on the rating scale.
  0;

type ValueOf<T> = T[keyof T];
const getMaxValue = (ratingScale: ValueOf<typeof RATING_SCALES>) => {
  switch (ratingScale) {
    case RATING_SCALES.zeroToFive:
      return 5;
    default:
      return 10;
  }
};

type Props = {
  ariaLabel: string,
  className?: string,
  ratingScale: ValueOf<typeof RATING_SCALES>,
  showScale?: boolean,
  size: ValueOf<typeof RATING_SIZES>,
  subtitle?: string,
  title?: string | ReactNode,
  value: string | number,
};

const BpkRating = (props: Props) => {
  const {
    ariaLabel,
    className,
    ratingScale,
    showScale,
    size,
    subtitle,
    title,
    value,
    ...rest
  } = props;

  const classNames = getClassName(
    'bpk-rating',
    className,
    size === RATING_SIZES.large && title && subtitle && 'bpk-rating--large',
  );
  const valueStyles = getClassName('bpk-rating__value');
  const scaleStyles = getClassName('bpk-rating__scale');
  const textWrapperStyles = getClassName(
    'bpk-rating__text-wrapper',
    size === RATING_SIZES.large && 'bpk-rating__text-wrapper--large',
  );
  const titleStyles = getClassName(
    subtitle && 'bpk-rating__title--with-subtitle',
  );
  const subtitleStyles = getClassName(
    'bpk-rating__subtitle',
  );

  let valueTextSize: ValueOf<typeof TEXT_STYLES> = TEXT_STYLES.label1;
  let scaleTextSize: ValueOf<typeof TEXT_STYLES> = TEXT_STYLES.caption;
  let titleTextSize: ValueOf<typeof TEXT_STYLES> = TEXT_STYLES.label1;
  let subtitleTextSize: ValueOf<typeof TEXT_STYLES> = TEXT_STYLES.caption;

  if (size === RATING_SIZES.large) {
    valueTextSize = TEXT_STYLES.hero5;
    scaleTextSize = TEXT_STYLES.bodyDefault;
    titleTextSize = TEXT_STYLES.heading5;
    subtitleTextSize = TEXT_STYLES.bodyDefault;
  }

  const minValue = getMinValue();
  const maxValue = getMaxValue(ratingScale);

  let adjustedValue = value;

  if(typeof adjustedValue === "number") {
    if (adjustedValue >= maxValue || adjustedValue <= minValue) {
      adjustedValue = clamp(adjustedValue, minValue, maxValue);
    }
  }

  return (
    <div className={classNames} aria-label={ariaLabel} role="figure" {...rest}>
      <div className={valueStyles}>
        <BpkText
          textStyle={valueTextSize}
          tagName="span"
          aria-hidden="true"
        >
          {adjustedValue}
        </BpkText>
        {showScale && (
          <BpkText
            textStyle={scaleTextSize}
            tagName="span"
            aria-hidden="true"
          >
            <span className={scaleStyles}>/{maxValue}</span>
          </BpkText>
        )}
      </div>

      <div className={textWrapperStyles}>
        {title && (
          <BpkText
            textStyle={titleTextSize}
            tagName="span"
            aria-hidden="true"
          >
            <span className={titleStyles}>{title}</span>
          </BpkText>
        )}

        {subtitle && (
          <BpkText
            textStyle={subtitleTextSize}
            tagName="span"
            aria-hidden="true"
          >
            <span className={subtitleStyles}>{subtitle}</span>
          </BpkText>
        )}
      </div>
    </div>
  );
};

BpkRating.propTypes = {
  ariaLabel: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  className: PropTypes.string,
  ratingScale: PropTypes.oneOf(Object.keys(RATING_SCALES)),
  size: PropTypes.oneOf(Object.keys(RATING_SIZES)),
  subtitle: PropTypes.string,
  showScale: PropTypes.bool,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
};

BpkRating.defaultProps = {
  className: null,
  ratingScale: RATING_SCALES.zeroToFive,
  size: RATING_SIZES.base,
  subtitle: null,
  showScale: true,
  title: null,
};

export default BpkRating;
