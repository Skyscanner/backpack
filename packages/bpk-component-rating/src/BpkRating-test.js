/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2019 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow strict */

import React from 'react';
import renderer from 'react-test-renderer';

import BpkRating from '../index';

describe('BpkRating', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <BpkRating
          ariaLabel="6.7 Average might recommend"
          title="Average"
          subtitle="Might recommend"
          value={6.7}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should support custom class names', () => {
    const tree = renderer
      .create(
        <BpkRating
          ariaLabel="6.7 Average might recommend"
          title="Average"
          subtitle="Might recommend"
          value={6.7}
          className="custom-classname"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a low score rating component', () => {
    const tree = renderer
      .create(
        <BpkRating
          ariaLabel="1.3 Low bad option"
          title="Low"
          subtitle="Bad option"
          value={1.3}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a medium score rating component', () => {
    const tree = renderer
      .create(
        <BpkRating
          ariaLabel="7 Average it's ok at best"
          title="Average"
          subtitle="It's ok at best"
          value={7}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a high score rating component', () => {
    const tree = renderer
      .create(
        <BpkRating
          ariaLabel="9 Amazing best thing since sliced bread"
          title="Amazing"
          subtitle="Best thing since sliced bread"
          value={9}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should correctly handling values lower than 0', () => {
    const tree = renderer
      .create(
        <BpkRating
          ariaLabel="-1.3 Low bad option"
          title="Low"
          subtitle="Bad option"
          value={-1.3}
          className="custom-classname"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should correctly handling values higher than 10', () => {
    const tree = renderer
      .create(
        <BpkRating
          ariaLabel="15 Amazing brilliant"
          title="Amazing"
          subtitle="Brilliant"
          value={15}
          className="custom-classname"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
