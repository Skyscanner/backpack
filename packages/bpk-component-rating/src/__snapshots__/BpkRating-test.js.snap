// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkRating should correctly handling values higher than 10 1`] = `
<div
  aria-label="15 Amazing brilliant"
  className="bpk-rating custom-classname"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--base-rating bpk-rating--high-rating"
  >
    <strong>
      10
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Amazing
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      Brilliant
    </span>
  </div>
</div>
`;

exports[`BpkRating should correctly handling values lower than 0 1`] = `
<div
  aria-label="-1.3 Low bad option"
  className="bpk-rating custom-classname"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--base-rating bpk-rating--low-rating"
  >
    <strong>
      0
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Low
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      Bad option
    </span>
  </div>
</div>
`;

exports[`BpkRating should render a high score rating component 1`] = `
<div
  aria-label="9 Amazing best thing since sliced bread"
  className="bpk-rating"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--base-rating bpk-rating--high-rating"
  >
    <strong>
      9.0
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Amazing
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      Best thing since sliced bread
    </span>
  </div>
</div>
`;

exports[`BpkRating should render a large rating component 1`] = `
<div
  aria-label="9 Excellent would recommend"
  className="bpk-rating"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--lg-rating bpk-rating--high-rating"
  >
    <strong>
      9.0
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--lg bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Excellent
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--horizontal"
    >
      This place was amazing
    </span>
  </div>
</div>
`;

exports[`BpkRating should render a low score rating component 1`] = `
<div
  aria-label="1.3 Low bad option"
  className="bpk-rating"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--base-rating bpk-rating--low-rating"
  >
    <strong>
      1.3
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Low
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      Bad option
    </span>
  </div>
</div>
`;

exports[`BpkRating should render a medium score rating component 1`] = `
<div
  aria-label="7 Average it's ok at best"
  className="bpk-rating"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--base-rating bpk-rating--medium-rating"
  >
    <strong>
      7.0
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Average
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      It's ok at best
    </span>
  </div>
</div>
`;

exports[`BpkRating should render a small rating component 1`] = `
<div
  aria-label="9 Excellent would recommend"
  className="bpk-rating"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--sm-rating bpk-rating--high-rating"
  >
    <strong>
      9.0
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Excellent
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      This place was amazing
    </span>
  </div>
</div>
`;

exports[`BpkRating should render correctly 1`] = `
<div
  aria-label="6.7 Average might recommend"
  className="bpk-rating"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--base-rating bpk-rating--medium-rating"
  >
    <strong>
      6.7
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Average
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      Might recommend
    </span>
  </div>
</div>
`;

exports[`BpkRating should render vertical rating correctly 1`] = `
<div
  aria-label="6.7 Average might recommend"
  className="bpk-rating bpk-rating--vertical"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--base-rating bpk-rating--medium-rating"
  >
    <strong>
      6.7
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--vertical"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--vertical"
    >
      <strong>
        Average
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--vertical"
    >
      Might recommend
    </span>
  </div>
</div>
`;

exports[`BpkRating should support custom class names 1`] = `
<div
  aria-label="6.7 Average might recommend"
  className="bpk-rating custom-classname"
>
  <span
    aria-hidden="true"
    className="bpk-text bpk-text--base bpk-rating__component bpk-rating--base-rating bpk-rating--medium-rating"
  >
    <strong>
      6.7
    </strong>
  </span>
  <div
    className="bpk-rating__text-wrapper bpk-rating__text-wrapper--horizontal"
  >
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--base bpk-rating__text bpk-rating__text--horizontal"
    >
      <strong>
        Average
      </strong>
    </span>
    <span
      aria-hidden="true"
      className="bpk-text bpk-text--sm bpk-rating__text bpk-rating__text--horizontal"
    >
      Might recommend
    </span>
  </div>
</div>
`;
